include/game/systems/placement_system.h

```diff
@@ -37,6 +43,46 @@ void updatePlacementSystem(entt::registry& mainRegistry, entt::registry& buildin
         if (selectedBuildingTemplate != entt::null)
         {
             // TODO check and spend resource
+            {
+                bool resourcesSufficient = true;
+
+                unsigned int woodCount = UI->getState()->woodCount;
+                unsigned int stoneCount = UI->getState()->stoneCount;
+
+                const auto& cost = buildingTemplateRegistry.try_get<Cost>(selectedBuildingTemplate);
+
+                if (cost != nullptr)
+                {
+                    if (cost->resource.uuid == "lumber")
+                    {
+                        if (cost->amount > woodCount)
+                        {
+                            resourcesSufficient = false;
+                        }
+                        else
+                        {
+                            UI->getMutableState()->woodCount = woodCount - cost->amount;
```

Comment:
This should be stored in the registry on A "Player" entity that has a Storage component for each resource

Reply:
You will have the same problem here "how to have multiple" resources
I you might an std::array or std::vector for that for now, since performance really is no issue so far.